test/widgets/test_check_updates.py

```diff
@@ -72,3 +76,115 @@ def test_no_update_available_with_no_update_string_and_color_no_updates(fake_qti
     assert cu4.layout.colour == cu4.colour_no_updates
+
+
+def test_update_available_with_restart_indicator(monkeypatch, fake_qtile):
+    """ test output with no indicator where restart needed """
+    cu5 = CheckUpdates(distro=good_distro,
+                       custom_command=cmd_1_line,
+                       restart_indicator="*",
+                       )
+    monkeypatch.setattr("os.path.exists", lambda x: True)
+    fakebar = Bar([cu5], 24)
+    fakebar.window = FakeWindow()
+    fakebar.width = 10
+    fakebar.height = 10
+    fakebar.draw = no_op
+    cu5._configure(fake_qtile, fakebar)
+    text = cu5.poll()
+    assert text == "Updates: 1*"
+
+
+def test_update_available_with_execute(manager_nospawn, minimal_conf_noscreen, tmpdir):
+    """ test polling after executing command """
+
+    # To check the execute function we need to simulate having updates before but none after
+    # We do this by having a file with a 'counter' in it. The check command reads the file,
+    # decreases the counter and, if the counter is greather than 0, returns a text string.
+    # When the counter is zero, an empty string is returned so we can check for the
+    # 'no_update_string.
+    dpath = tmpdir.dirpath
+    counter_file = str(dpath("counter.txt"))
+    cmd_counter = 'export COUNT=$(cat {path}); echo $((--COUNT)) > {path}; [ "$COUNT" = "0" ] || echo "Updates"'
+    cmd_counter = cmd_counter.format(path=counter_file)
+
+    # Start with a value of 2. On the first run, the counter will be 1, on the second it will be 0
+    with open(counter_file, "w") as cfile:
+        cfile.write("2")
+
+    # Simulate a short process for the execute command as we want to test the widget's looping while
+    # the execute command is running
+    cu6 = CheckUpdates(distro=good_distro,
+                       custom_command=cmd_counter,
+                       execute="sleep 0.1",
+                       no_update_string=nus,
+                       )
+    # Shorten the default gap between
+    cu6.execute_polling_interval = 0.02
+
+    config = minimal_conf_noscreen
+    config.screens = [
+        libqtile.config.Screen(
+            top=libqtile.bar.Bar([cu6], 10)
+        )
+    ]
+
+    manager_nospawn.start(config)
+
+    topbar = manager_nospawn.c.bar["top"]
+
+    assert topbar.info()["widgets"][0]["text"] == "Updates: 1"
+
+    # Clicking the widget triggers the execute command
+    topbar.fake_button_press(0, "top", 0, 0, button=1)
+
+    # Ugly solution but we just loop until the text changes or we time out
+    # The ONLY way the text can change here is if the check_updates command
+    # is triggered after the completion of the execute command (the default
+    # polling interval is too large to trigger an update here)
+    retries = 10
+    while topbar.info()["widgets"][0]["text"] != nus and retries > 0:
+        sleep(0.05)
+        retries -= 1
+
+    # Assert that we didn't time out and the execute command has been run
+    assert retries > 0
```

Comment:
This works on my system but not in the CI testing environment.  The test is failing because it times out. 

Any suggestions on how to redo/fix this test?

Reply:
Actually, I've got another idea. Will test later.